### DISCUSSION TITLE
Enabling middleware conditionally for some builds only?

### DISCUSSION THREAD

**Discussion by tbgse** [vercel/next.js#39747] [Help]
Hi all,

I was wondering if there is a way to only run middleware on the Vercel preview builds / conditionally depending on environment variables but to disable it entirely in production? I've already tried dynamically setting matcher routes, but that doesn't seem to be an option, since they can't be configured. 

Anyone found a solution to this already? It'd be really nice to have more control over this in each build environment.

**Comment by kochis** [comment]
I'm running into the same problem. Curious if you come up with a solution for this.

**Comment by im-cho** [comment]
same here. i needed to implement basic auth in test environment but not in the live environment. but i can't seem to find a way to run the middleware only in specific environment. wondering if anyone has experience with this?

**Reply by tillschweneker** [reply]
I'm using a pre-build script to conditionally rename the file middleware__DISABLED.ts to middleware.ts. The script only runs when a specific environment variable is set. 

Not ideal, but it works to enable basic auth for specific environments, for example.

**Reply by im-cho** [reply]
> I'm using a pre-build script to conditionally rename the file middleware__DISABLED.ts to middleware.ts. The script only runs when a specific environment variable is set.
> 
> Not ideal, but it works to enable basic auth for specific environments, for example.

I think this will be my quick solution for now, thanks a lot!
